tests/CMakeLists.txt

```diff
@@ -180,6 +180,18 @@ IF(${PYTHON_TEST})
   set(PYTHON_TEST_HELPERS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/python)
 ENDIF()
 
+IF (ALBANY_ENABLE_CUDA AND NOT ALBANY_ENABLE_CUDA_UVM)
```

Comment:
This doesn't seem right, since it would skip landice tests on non-cuda platforms. If you're trying not to run tests with UVM on, you should simply to
```
if (NOT ALBANY_ENABLE_CUDA_UVM)
```

Reply:
It's a bit awkward but notice there's a return in this if-statement. If this specific condition isn't met, this block won't run and the old behavior will happen (including adding the landice tests later in a subsequent if-block). 

This seemed like the least intrusive way to enable the specific tests I want without adding `and not ALBANY_ENABLE_CUDA_UVM` to every if statement. I've confirmed the same tests run on CUDA builds, I will double check the same tests are enabled for a serial build.

Reply:
Ah, I missed that! So are we just trying to exclude code/demo if CUDA=ON and UVM=OFF, right? Then, since landice/unit subdirs are added anyways, maybe we should do something like

```
# Unit tests (always run)
if (ALBANY_UNIT_TESTS)
  add_subdirectory(unit)
endif()

# Do not run code/demo tests on GPU unless UVM is on
if (NOT ALBANY_ENABLE_CUDA OR ALBANY_ENABLE_CUDA_UVM)
  add_subdirectory(corePDEs)
  
  if (ENABLE_DEMO_PDES)
    add_subdirectory(demoPDEs)
  endif()
endif()

# Landice needs SEACAS
if(ENABLE_LANDICE AND ALBANY_SEACAS )
  add_subdirectory(landIce)
endif()
```

Reply:
Or you could still use return, but I would use it like this
```cmake
<add all subdirs that are added anyways>

if (ALBANY_ENABLE_CUDA AND NOT ALBANY_ENABLE_CUDA_UVM)
  return()
endif()

<add all remaining subdirs that are only executed if CUDA=OFF or UVM=ON
```

Reply:
Good call, I just updated the logic accordingly to reduce code duplication and be less confusing.